scantool/scantool_850.c

```diff
@@ -95,13 +95,454 @@ struct dtc_table_entry {
+	{0x7a, 0x53, 446, "Pulsed secondary air injection system valve leakage [P0410] [see OTP Volvo 850 DVD, TP-2308202, p669, p78, p344]"},
+	{0x7A, 0x54, 445, "Pulsed secondary air injection system pump signal [P0410] [see OTP 850 DVD, TP-2318201, p268 and p44]"},
+	{0x7A, 0x55, 447, "Pulsed secondary air injection system solenoid valve signal [P0413 (missing); P0414 (low/high)] [see OTP 850 DVD, TP-2318201, p268 and p44]"},
+	{0x7A, 0x56, 241, "EGR system"},
+	{0x7A, 0x59, 413, "EGR temperature sensor signal"},
+	{0x7A, 0x5D, 315, "Canister purge (CP) valve leakage"},
+	{0x7A, 0x5E, 611, "Fuel Tank system, large leakage [P0455]"},
+	{0x7A, 0x5F, 614, "EVAP canister shut off valve flow fault"},
+	{0x7A, 0x60, 616, "EVAP canister shut off valve signal"},
+	{0x7A, 0x61, 612, "Fuel Tank system, small leakage [P0442] [check fuel cap seated and sealed properly, leaking fuel tank, filler pipe, EVAP canister or its shut-off valve, lines connecting any of those, or bad fuel tank pressure sensor]"},
+	{0x7A, 0x62, 541, "Canister purge (CP) valve signal"},
+	{0x7A, 0x63, 621, "Fuel tank pressure sensor signal"},
+	{0x7A, 0x65, 112, "MFI Control Module (CM) fault"},
+	{0x7A, 0x67, 112, "Memory Fault (ROM) In Engine Control Module (ECM)"},
+	{0x7A, 0x6B, 132, "Battery Voltage"},
+	{0x7A, 0x6D, 315, "Canister purge (CP) valve leakage"},
+	{0x7A, 0x6E, 131, "RPM sensor signal missing"},
+	{0x7A, 0x6F, 214, "RPM sensor signal sporadic faulty"},
+	{0x7A, 0x70, 314, "Camshaft Position sensor (CMP) signal [P0340] [see OTP 850 DVD, TP-2308202, p262 for possible sources]"},
+	{0x7A, 0x73, 121, "Mass Air Flow (MAF) sensor signal"},
+	{0x7A, 0x75, 411, "Throttle Position (TP) sensor signal"},
+	{0x7A, 0x77, 422, "Atmospheric pressure sensor signal"},
+	{0x7A, 0x78, 311, "Engine RPM high for too long while no Vehicle Speed"},
+	{0x7A, 0x79, 355, "Mass air flow (MAF) sensor, faulty signal"},
+	{0x7A, 0x7A, 112, "Fault in Engine Control Module (ECM) NTC switch"},
+	{0x7A, 0x7B, 123, "Engine Coolant Temperature (ECT) sensor [P0116 (faulty), P0117 (low), P0118 (high), or P0115 (missing)]"},
+	{0x7A, 0x7C, 251, "Ambient Temperature sensor signal [P0111 (sporadic), P0112 (low), or P0113 (high)]"},
+	{0x7A, 0x7F, 112, "Fault in Engine Control Module (ECM) Temperature"},
+	{0x7A, 0x80, 432, "Temperature warning, level 1"},
+	{0x7A, 0x81, 513, "Temperature warning, level 2"},
+	{0x7A, 0x82, 225, "A/C pressure sensor signal"},
+	{0x7A, 0x96, 115, "Injector 1"},
+	{0x7A, 0x97, 125, "Injector 2"},
+	{0x7A, 0x98, 135, "Injector 3"},
+	{0x7A, 0x99, 145, "Injector 4"},
+	{0x7A, 0x9A, 155, "Injector 5"},
+	{0x7A, 0x9B, 165, "Injector 6"},
+	{0x7A, 0xA4, 335, "Request for MIL lighting from TCM [or possibly Comm Failure between Motronic 4.4 ECM and AW 50-42 TCM] [P1618 (High), P1617 (Low/Missing/Faulty)] [check grounds 31/71, 31/32, 31/33; check TCM pin B15 to M44 pin B26 circuit & connections] [see OTP 850 DVD: TP-2308032, pp275-280; TP-2318201, p43, p269]"},
```

Comment:
I'm sure that 0x1D is EFI-666 DTC in TCM / MIL request from TCM. So 0xA4 EFI-335 must not be MIL request from TCM - it must be cable fault between TCM and ECM (the MIL request signal is shorted or open).